Add sieve protocol support when using STARTTLS

k0lter opened the pull request:
This PR add support for the (manage)sieve protocol while using STARTTLS (if the underlying openssl supports it).

k0lter committed:
Add (manage)sieve protocol support when using STARTTLS

drwetter commented:
Thank you! 

Basically a good idea, we need to extend that a bit though.

* Is there any public server you know which we can use for unit testing?
* if not: is there any server you know about where I can privately test it myself?
* ``prepare_debug()`` misses a HAS_SIEVE line
* the "reasonable port guess function" in ``ports2starttls()``: can that be amended?
* documentation is missing in ~/doc/

drwetter commented:
désolé, pushed da wrong button (David Guetta)

k0lter committed:
Fix ports2starttls() to support (manage)sieve protocol while using STARTTLS

k0lter committed:
Add unit testing (manage)sieve protocol while using STARTTLS

k0lter committed:
Update doc for (manage)sieve protocol when used with STARTTLS

k0lter commented:
I've updated the PR to take into account your proposals:

- Unit test added with a known public imap server which supports (manage)sieve through STARTTLS
- `ports2starttls()` was updated
- doc (markdown, manpage and html) also updated
- `HAS_SIEVE` was already present in `prepare_debug()` so no changes

Note: the OpenSSL binary present in `bin/` (at least the Linux amd64 one) directory seems to be based on OpenSSL 1.0.2 but the sieve protocol support with STARTTLS was added in OpenSSL version 1.1.1

drwetter commented:
Well done, thanks a lot! 

I would like to merge this. However the roff-formatted man page and the HTML introduce more changes than just adding the word sieve. Probably this is due to the use of ronn-ng which is admittedly better. 

But despite that: let's move forward. There's one thing  though which each version of ronn doesn't do good: It limits the width for HTML to 100 pixels . If you could just change that?

drwetter commented on doc/testssl.1.html:
```diff
@@ -1,13 +1,13 @@
   /* style: man */
   body#manpage {margin:0}
-  .mp {padding:0 9ex 1ex 4ex}
+  .mp {max-width:100ex;padding:0 9ex 1ex 4ex}
```
This line.

k0lter replied:
I was about to fix this but I found some inconsistencies in doc so I've submitted the PR #2074.

Could you take a look and if you merge it I could update this PR accordingly.

drwetter replied:
FYI: I fixed this one manually now in another PR

drwetter commented:
Thanks!

Your other PR is now colliding with this one though.
